app/src/main/java/com/canvearth/canvearth/server/FBPixelManager.java

```diff
@@ -208,8 +232,8 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
         return bitmap;
     }
 
-    public void writePixelAsync(PixelData pixelData, PixelColor pixelColor, @Nullable Function<PixelData> callback) {
-        new Thread(() -> {
+    public Thread writePixelAsync(PixelData pixelData, PixelColor pixelColor) {
+        Thread thread = new Thread(() -> {
```

Comment:
How about using a Handler like Moogung suggested?

Reply:
Since we are planning to move this execution to server, I think preventing this callback is better for our future.
Thus, I don't really think this callback is needed, since we can do same work by event occurrence of watching pixel. (and I think it's better.)
However, if you really need this callback, I will modify this. Tell me when you need.

Reply:
Oh sorry I didn't mean brining the callback back. I meant to suggest using a ThreadPoolExecutor as in https://developer.android.com/training/multiple-threads/run-code.

Reply:
Got it